layers/dns.go

```diff
@@ -816,6 +842,29 @@ func decodeCharacterStrings(data []byte) ([][]byte, error) {
 	return strings, nil
 }
 
+func decodeOPTs(data []byte, offset int) ([]DNSOPT, error) {
+	allOPT := []DNSOPT{}
+	end := len(data)
+
+	if offset == end {
+		return allOPT, nil // There is no data to read
+	}
+
+	if offset+4 > end {
+		return allOPT, fmt.Errorf("DNSOPT record is of length %d, it should be at least length 4", end-offset)
+	}
+
+	for i := offset; i < end; {
+		opt := DNSOPT{}
+		opt.Code = DNSOptionCode(binary.BigEndian.Uint16(data[offset : offset+2]))
+		l := binary.BigEndian.Uint16(data[offset+2 : offset+4])
+		opt.Data = data[offset+4 : offset+4+int(l)]
```

Comment:
This is missing a length check

Reply:
Added the length check and a corresponding test in dns_test.go